app/controllers/halls_controller.rb

```diff
@@ -1,44 +1,47 @@
 # frozen_string_literal: true
 
 class HallsController < ApplicationController
-
   def index
-    halls = Hall.all
-    render json: HallSerializer.new(halls)
+    hall = repo.all
+    render json: HallSerializer.new(hall)
```

Comment:
You can change to `render jsonapi: hall` and it will automatically find good serializer.